subsys/usb/usb_transfer.h

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) 2019 PHYTEC Messtechnik GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_USB_TRANSFER_H_
+#define ZEPHYR_USB_TRANSFER_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Initialize USB transfer data
+ *
+ * @return 0 on success, negative errno code on fail.
+ */
+int usb_transfer_init(void);
```

Comment:
So we have this header only for `usb_transfer_init()` and other header for `usb_transfer{,_sync}()`

Reply:
Yes, this is internal for device stack, [include/usb/usb_device.h](https://github.com/zephyrproject-rtos/zephyr/blob/master/include/usb/usb_device.h#L351) is public API for users.

Reply:
Probably `usb_transfer.h` can be merged with `os_desc.h` and `usb_descriptor.h` to a `usb_device_internal.h`

Reply:
I think it would be better to first merge this PR and then think about merge you mentioned in order to resolve this #20236